### ISSUE TITLE
Potential bug related to Lint/AmbiguousBlockAssociation

### ISSUE THREAD

**Issue by apainintheneck** [rubocop/rubocop#10451]
I expected the following code to not get flagged by the Lint/AmbiguousBlockAssociation cop. The only thing I can think of is that the trailing if statement might be confusing the cop. It could also be that this is the desired behavior and I'm just not aware.

## Expected behavior

I expect the following code to pass without any warnings.

```Ruby
assert MachO::LoadCommands::SEGMENT_FLAGS.keys.one? { |sf| seg.flag?(sf) } if seg.flags != 0
```
## Actual behavior

>  W: Lint/AmbiguousBlockAssociation: Parenthesize the param
> MachO::LoadCommands::SEGMENT_FLAGS.keys.one? { |sf| seg.flag?(sf) } 
> to make sure that the block will be associated with the 
> MachO::LoadCommands::SEGMENT_FLAGS.keys.one? method call.

## Steps to reproduce the problem

Run the Lint/AmbiguousBlockAssociation cop on the above code snippet.

## RuboCop version

```
$ bundle exec rubocop -V
1.25.1 (using Parser 3.1.1.0, rubocop-ast 1.16.0, running on ruby 3.1.1 x86_64-darwin19)
```
